test/test_lexer.ml

```diff
@@ -20,10 +20,35 @@ let next_token_tests =
          make_next_token_test "empty program" "" [ EOF ];
          make_next_token_test "program with whitespace" "   \n\r\t\n" [ EOF ];
          make_next_token_test "basic assignment expression" "const x = 1 + 2;"
-           [ Const; Iden "x"; Equal; Int 1; Add; Int 2; Semi; EOF ];
+           [ Const; Iden "x"; Assign; Int 1; Add_Subtract; Int 2; Semi; EOF ];
+         make_next_token_test "read plus minus" "+ ++ - -- += -= +9 -test"
+           [
+             Add_Subtract;
+             Inc_Dec;
+             Add_Subtract;
+             Inc_Dec;
+             Assign;
+             Assign;
+             Add_Subtract;
+             Int 9;
+             Add_Subtract;
+             Iden "test";
+             EOF;
+           ];
+         make_next_token_test "basic assignment expression" "= == === ==== =>"
+           [ Assign; Equality; Equality; Equality; Assign; Arrow; EOF ];
          make_next_token_test "compact assignment expression"
            "const variable=1+2;"
-           [ Const; Iden "variable"; Equal; Int 1; Add; Int 2; Semi; EOF ];
+           [
+             Const;
+             Iden "variable";
+             Assign;
+             Int 1;
+             Add_Subtract;
```

Comment:
Did this based on other lexers I saw, but feel free to argue against it. We could do `Additive` instead or keep them separate `Add` and `Subtract` whatever works for me.